hardhat.config.js/index.js

```diff
@@ -21,6 +21,14 @@ const config = {
     apiKey: process.env.ETHERSCAN_API_KEY,
   },
 
+  typechain: {
+    outDir: 'src/types',
+    target: 'ethers-v5',
+    alwaysGenerateOverloads: false,
+    externalArtifacts: ['externalArtifacts/*.json'],
+    dontOverrideCompile: true, // defaults to false
```

Comment:
- `outDir` should be `'types'` (drop the `src/`)
- we currently don't have external artifacts, remove that line. if we want it - let's do another PR as it's a new feature that's out of scope for this one